src/objective_coefficients.jl

```diff
@@ -0,0 +1,103 @@
+# Primals
+"""
+        PrimalObjectiveCoefficients{T}
+
+Primal objective coefficients defined as ``a_0^Tx + b_0`` as in
+http://www.juliaopt.org/MathOptInterface.jl/stable/apimanual/#Advanced-1
+
+`affine_terms` corresponds to ``a_0`` 
+`vi_vec` corresponds to ``x`` 
+`constant` corresponds to ``b_0`` 
+"""
+struct PrimalObjectiveCoefficients{T}
+    affine_terms::Vector{T}
+    vi_vec::Vector{VI}
+    constant::T
+end
+
+const POC{T} = PrimalObjectiveCoefficients{T}
+
+"""
+        get_POC(model::MOI.ModelLike)
+
+Get the coefficients from the primal objective function and
+return a `PrimalObjectiveCoefficients{T}`
+"""
+function get_POC(model::MOI.ModelLike)
+    return _get_POC(model, model.objective)
+end
+
+function _get_POC(model::MOI.ModelLike, obj_fun::MOI.ScalarAffineFunction{T}) where T
+    # Empty vector a0 with the number of variables
+    num_terms = length(obj_fun.terms)
+    a0 = Vector{T}(undef, num_terms)
+    vi = Vector{VI}(undef, num_terms)
+    # Fill a0 for each term in the objective function
+    i = 1
+    for term in obj_fun.terms
+        a0[i] = term.coefficient # scalar affine coefficient
+        vi[i] = term.variable_index # variable_index
+        i += 1
+    end
+    b0 = model.objective.constant # Constant term of the objective function
```

Comment:
Isn't this `obj_fun.constant` ? In that case `model` is not needed in this function